### PR TITLE
Date range default value

### LINKED ISSUE #1984: Default date range for DateRange input
Ability to set a default date range which will take effect on page load.

e.g., defaultValue=“Last 7 days”

### PR DESCRIPTION
**Pull request by Franreno**
### Description

This PR aims to add a defaultValue parameter to the DateRange input component.
The defaultValue has the purpose to define a default range when the component is first loaded.

To use is to simply add the defaultValue as below:
```js
<DateRange 
    data={orders_by_month} 
    dates=month 
    name=range_default 
    defaultValue="Last 90 days"
/>
```
Then, the DateRange will initialize with the defaultValue provided

Screenshot 2024-05-08 at 19 14 02

If the parameter is not provided or does not comply with the labels available on the `preset` variable, then the component's behavior should be the same as previously

Screenshot 2024-05-08 at 19 13 52

Closes #1984

### Checklist

- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after
- [X] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] I have added to the docs where applicable
- [ ] I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable

### CODE REVIEW

File: .changeset/early-crews-compete.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@evidence-dev/core-components': patch
+'@evidence-dev/components': patch
+---
+
+Add defaulValue to DateRange
```
Comment (ItsMeBrianD) [inline .changeset/early-crews-compete.md:6]:
```suggestion
Add defaultValue to DateRange
```

Comment (Franreno) [reply .changeset/early-crews-compete.md:6]: Ops, thanks :D

**Review by mcrascal — CHANGES_REQUESTED** [review]
This is awesome, I like this string based API for selecting default date ranges. 

The only thing that seems off to me here is the loading behaviour. It seems that in the process of loading, results from queries that depend on the value here transition through an empty state, causing a no data error to be briefly visible in dependent components. @ItsMeBrianD can probably help identify the issue. 

CleanShot 2024-05-17 at 10 56 17

**Comment by ItsMeBrianD** [comment]
> The only thing that seems off to me here is the loading behaviour. It seems that in the process of loading, results from queries that depend on the value here transition through an empty state, causing a no data error to be briefly visible in dependent components. @ItsMeBrianD can probably help identify the issue.

I think this was fixed in `features/b`, merging next in should resolve the issue

**Comment by archiewood** [comment]
@kwongz was this addressed by your PR? Can we close this?

**Comment by kwongz** [comment]
@archiewood  Yup! Default Values are now available as of the latest updates

**Pull request closed without merging**